credsweeper/deep_scanner/html_scanner.py

```diff
@@ -19,7 +19,7 @@ def data_scan(
             depth: int,  #
             recursive_limit_size: int) -> List[Candidate]:
         """Tries to represent data as html text and scan as text lines"""
-        if data_provider.represent_as_html():
+        if data_provider.represent_as_html(self.keyword_check, depth, recursive_limit_size):
```

Comment:
It seems unnatural passing `self.keyword_check` method here...
As I see, `represent_as_html()` method just check if the data can be parsed as html or not.
What about leave `represent_as_html()` as it is and call `keyword_check()` inside of `html_scanner.py`?

Reply:
The represent method does not only check but perform the representation.
The keyword_check is passed to reduce code duplication. It helps also skip table headers combination when a header does not contain required substring.

Reply:
Hmm..
Checking `required_substrings` already existed in `Scanner` class.
https://github.com/Samsung/CredSweeper/blob/1bf1d1f64372ac9adb576e9aba1d98f9e29d3c8d/credsweeper/scanner/scanner.py#L137-L142

Is it really needed to be checked on `HTMLScanner` class?
Maybe it can increase the performance.. but I think it makes code more complicated.